### ISSUE TITLE
File locking

### ISSUE THREAD

**Issue by guedressel** [rfjakob/gocryptfs#39]
Does gocryptfs do file locking through the fuse layer?

So if a mounted & decrypted file gets a lock by a process using it, will the underlying encrypted source file also have a lock?

**Comment by rfjakob** [comment]
No, unfortunately file locking requests are currently thrown away by the go-fuse library. This is mentioned in the [go-fuse README](https://github.com/hanwen/go-fuse/blob/master/README) as a TODO, so there are good chances of getting it merged.

**Comment by rfjakob** [comment]
Ok to close I guess?

**Comment by guedressel** [comment]
Well - actually it's postponed, not closed.
To keep record on the go-fuse progress I created an issue there: hanwen/go-fuse#134

**Comment by guedressel** [comment]
hanwen/go-fuse#134 got closed!

**Comment by rfjakob** [comment]
Nice!

**Comment by rfjakob** [comment]
Unfortunately, the Flock implementation in go-fuse does not seem to be working yet: https://github.com/hanwen/go-fuse/pull/153#issuecomment-298732540

**Comment by rfjakob** [comment]
Just curious: What is your use case for this? I'm afraid this is going to take a while to land in go-fuse.

**Comment by charles-dyfis-net** [comment]
This isn't my ticket originally, but my own use case is locking files on a NFS store multiple systems are accessing via gocryptfs.

**Comment by guedressel** [comment]
My use case is to have the underlying encrypted backing files being synced across multiple computers with sync tools (syncany, dropbox, ...).
If an unencrypted file is in use / has a lock it's underlying encrypted file might still be overwritten by such a tool if it can't see the lock.

**Comment by lf-zbw** [comment]
https://github.com/hanwen/go-fuse/pull/220 seems fixed flock, could you have a look at it? Thanks!

**Comment by kototama** [comment]
Any news on these? If I understand correctly it prevents gocryptfs to be used with cloud sync tools.

**Comment by rfjakob** [comment]
No news here, but this is not related to cloud sync tools.

Dropbox, nextcloud etc work fine

**Comment by rfjakob** [comment]
(tbh, i doubt that dropbox etc even look at file locks)

**Comment by kototama** [comment]
Oh ok, I misunderstood the issue in that case. Thanks for the quick answer.

**Comment by kototama** [comment]
Out of topic but I don't know where to ask: so will gocryptfs avoid corruption when two cloud sync clients modify the same file on two computers at the same time?

**Comment by rfjakob** [comment]
What usually happens is that the cloud sync client creates a file like

    my file (copy 1).txt
